File: UNP/ref/Python/TADs e Classes/pilha_estatica.py
class Pilha(object):
    def __init__(self, max):
        self.dados = [None] * max
        self.topo = -1

    def empilha(self, elemento):
        if self.topo < len(self.dados) - 1:
            self.topo += 1
            self.dados[self.topo] = elemento

    def desempilha(self):
        if self.topo >= 0:
            elemento = self.dados[self.topo]
            self.topo -= 1
            return elemento

    def vazia(self):
        return self.topo == -1
    
    def peek(self):
        if not self.vazia():
            return self.dados[self.topo]
    
    def exibe_pilha_completa(self):
        print('Pilha completa: ', self.dados)

    def exibe_topo(self):
        print('Topo: ', self.peek())


pilha = Pilha(5)
pilha.empilha(1)
pilha.empilha(2)
pilha.exibe_topo()
pilha.empilha(3)
pilha.empilha(4)
pilha.exibe_topo()
pilha.empilha(5)
pilha.exibe_topo()
pilha.exibe_pilha_completa()

File: UNP/ref/Java/#ED2-Java-BKP/EstruturasBasicas/Pilha/PilhaEstatica.java
package EstruturasBasicas.Pilha;

import javax.print.PrintException;

public class PilhaEstatica {
  private Object[] elementos;
  private int topo;

  public PilhaEstatica(int maximo) {
    this.elementos = new Object[maximo];
    this.topo = -1;
  }

  public boolean estaVazia() {
    return this.topo == -1;
  }

  public boolean estaCheia() {
    return this.topo == this.elementos.length - 1;
  }

  public void empilha(Object elemento) throws PrintException {
    if (this.estaCheia()) {
      throw new PrintException("A pilha está cheia");
    }
    this.elementos[++topo] = elemento;
  }

  public Object desempilha() throws PrintException {
    if (this.estaVazia()) {
      throw new PrintException("A pilha está vazia");
    }
    return this.elementos[topo--];
  }

  public Object topo() {
    if (this.estaVazia()) {
      return null;
    }
    return this.elementos[topo];
  }

  public int tamanho() {
    return this.topo + 1;
  }

  public void imprime() {
    if (this.estaVazia()) {
      System.out.println("Pilha vazia");
    } else {
      for (int i = 0; i <= this.topo; i++) {
        System.out.println(this.elementos[i]);
      }
    }
  }

  public static void main(String[] args) throws PrintException {
    PilhaEstatica pilha = new PilhaEstatica(10);
    pilha.empilha("Edkallenn");
    pilha.empilha(4);
    pilha.empilha("Paulo");
    pilha.empilha("Rafael");
    pilha.empilha("Ana");
    pilha.empilha("João");
    pilha.empilha("Maria");
    pilha.empilha("José");
    pilha.empilha("Carlos");
    pilha.empilha("Edson");
    pilha.imprime();
    System.out.println("Desempilhando: " + pilha.desempilha());
    System.out.println("Desempilhando: " + pilha.desempilha());
    System.out.println("Desempilhando: " + pilha.desempilha());
    System.out.println("Desempilhando: " + pilha.desempilha());
    System.out.println("Desempilhando: " + pilha.desempilha());
    System.out.println("Desempilhando: " + pilha.desempilha());
    System.out.println("Desempilhando: " + pilha.desempilha());
    System.out.println("Desempilhando: " + pilha.desempilha());
    System.out.println("Desempilhando: " + pilha.desempilha());
    pilha.imprime();
  }
}
